tests/framework/installer/ceph_installer.go

```diff
@@ -477,7 +480,17 @@ func (h *CephInstaller) purgeClusters() error {
 func (h *CephInstaller) checkCephHealthStatus(namespace string) {
 	clusterResource, err := h.k8shelper.RookClientset.CephV1().CephClusters(namespace).Get(namespace, metav1.GetOptions{})
 	assert.Nil(h.T(), err)
-	assert.Equal(h.T(), "Created", string(clusterResource.Status.State))
+
+	for name, message := range clusterResource.Status.CephStatus.Details {
+		logger.Infof("Ceph health message before checking clusterState: %s. %s: %s", name, message.Severity, message.Message)
+	}
+
+	// Let's wait for the OSDs to reconcile
+	clusterState := string(clusterResource.Status.State)
+	if clusterState != "Created" {
+		time.Sleep(600 * time.Second)
```

Comment:
Adding 10m to the integration tests is a pretty steep penalty. Can we have a short timeout such as 5s and retry for a few minutes? You would also need to query the status again. Even with this sleep, the assert will fail on the next line since `clusterState` is not updated. But maybe we don't need this wait anymore since you said the upgrade will not be blocked on a single node?

Reply:
Removed.